src/modules/variable/init.rs

```diff
@@ -39,23 +41,39 @@ impl SyntaxModule<ParserMetadata> for VariableInit {
         // Get the variable name
         let tok = meta.get_current_token();
         self.name = variable(meta, variable_name_extensions())?;
+
+        if let Ok(_) = token(meta, ":") {
+            // matches declarations w/o assigning value:
+            // let name: Text
+            let typee = parse_type(meta)?;
+            self.handle_add_variable(meta, &self.name.clone(), typee, tok, true)?;
+            self.is_declare = true;
+            return Ok(());
```

Comment:
Why cant we allow for this syntax?

```
let x: Text = fun_returning_any()
```

Reply:
we can, its just not relevant in this PR. you should create another issue for that